php-checks/src/test/resources/checks/UnusedLocalVariableCheck.php

```diff
@@ -281,3 +281,19 @@ function extract_function() {
   extract($var_array);
   echo "$color";
 }
+
+/**
+ * SONARPHP-760
+ */
+function anonymous_class() {
+
+  $foo760 = '...'; // used in anonymous class
+  new class($foo760) {
+    private $bar;
+    public function __construct($bar) { $this->bar = $bar; }
+  };
+
+  $bar760 = '...'; // used in anonymous class
+  new class($bar760) { };
```

Comment:
This is a FN, should I check here that the anonymous class has a constructor with N parameters, where N is the number of variables passed to the class as parameters?
It would add complexity to the implementation...